cupyx/scipy/sparse/compressed.py

```diff
@@ -433,6 +436,85 @@ def _with_data(self, data, copy=True):
                 shape=self.shape,
                 dtype=data.dtype)
 
+    def prune(self):
+        """Remove empty space after all non-zero elements.
+        """
+        major_dim = self._swap(*self.shape)[0]
+
+        if self.indptr.size != major_dim + 1:
+            raise ValueError('index pointer has invalid length')
+
+        if self.indices.size < self.indptr[-1]:
```

Comment:
Please define `new_nnz` and reuse it to reduce device to host synchronization.
```suggestion
        new_nnz = int(self.indptr[-1])
        if self.indices.size < new_nnz:
```